components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -157,6 +157,11 @@ func (g *rtkSerial) Reconfigure(ctx context.Context, deps resource.Dependencies,
 		NtripConnectAttempts: newConf.NtripConnectAttempts,
 	}
 
+	last := len(ntripConfig.NtripURL) - 1
+	if ntripConfig.NtripURL[last] == '/' {
+		ntripConfig.NtripURL = ntripConfig.NtripURL[:last]
+	}
+
```

Comment:
So this is a workaround, but doesn't fix the actual bug, which I believe is in gpsutils/vrs.go:ConnectToVirtualBase(). In there, it's blindly stirpping prefixes off the URL as string operations, rather than running it through url.parse and taking the host:port from that parsed URL.

I think it'd be much better to fix the source of the bug instead of just covering it up in this one case. If I accidentally put any OTHER character than a / on the end of the URL (like a stray space or colon) this would break again.

It might also be good to update the Validate() function to actually parse the URL, instead of just checking that it's non-empty, so the user gets a clear message for actually invalid URLs too.